### PR TITLE
Feature/tracks

### PR DESCRIPTION
**Pull request by mindovermiles262**
Basic Implementation of Tracks feature.

- [x] Add Tracks Model
- [x] Add TracksCourses Join Table
- [x] Configure seeds file to create first Tracks
- [x] Add `/track` routes
- [x] Build `TracksController`
- [x] Add Track to User
- [ ] Create track selection dropdown in `Tracks#Show` view
- [ ] Stylize `Tracks#Index` and `Tracks#Show` Views
- [ ] Parameterize URL strings
- [ ] Edit navbar to redirect to `/tracks` url
- [ ] Add track choice to user's dashboard

### CODE REVIEW

**Comment by KevinMulhern** [comment]
Looks Good so far @mindovermiles262, It could use some unit tests though 😄

**Comment by mindovermiles262** [comment]
Incorrect model/track_course join table. Will open new PR when ready

**Comment by mindovermiles262** [comment]
Reopened and merged so @KevinMulhern can branch from TOP/feature/tracks branch to write tests for this PR.

File: app/models/course.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@ class Course < ApplicationRecord
 
   has_many :sections, -> { order(:position) }
   has_many :lessons, through: :sections
+  has_and_belongs_to_many :tracks
```
Comment (arku) [inline app/models/course.rb]: @mindovermiles262 Curious to know why we ended up with `has_and_belongs_to_many` here.

File: app/models/user.rb [hunk 2]
```diff
@@ -15,6 +15,7 @@ class User < ApplicationRecord
   has_many :completed_lessons, through: :lesson_completions, source: :lesson
   has_many :projects, dependent: :destroy
   has_many :user_providers, dependent: :destroy
+  belongs_to :track
```
Comment (arku) [inline app/models/user.rb]: @mindovermiles262 This makes an assumption a user is only going to be associated with a single track. Is that a right assumption ? :thinking:

**Pull request merged**